src/Components/Blazor/Blazor/src/Http/WebAssemblyHttpMessageHandler.cs

```diff
@@ -114,10 +119,25 @@ public class WebAssemblyHttpMessageHandler : HttpMessageHandler
         private static byte[] AllocateArray(string length) => new byte[int.Parse(length)];
 #pragma warning restore IDE0051 // Remove unused private members
 
-        private string[][] GetHeadersAsStringArray(HttpRequestMessage request)
-            => (from header in request.Headers.Concat(request.Content?.Headers ?? Enumerable.Empty<KeyValuePair<string, IEnumerable<string>>>())
-                from headerValue in header.Value // There can be more than one value for each name
-                select new[] { header.Key, headerValue }).ToArray();
+        private static IReadOnlyList<Header> GetHeaders(HttpRequestMessage request)
```

Comment:
What? You don't like my code-golfing LINQ expressions??? Look how many more lines we have now!

Oh sure your code is much easier to understand and debug, but think of the extra disk space this consumes!